src/controllers/auth.contoller.js

```diff
@@ -0,0 +1,129 @@
+const { userSchema } = require('../models/user');
+
+const { authServices } = require('../services/auth.services');
+const { jwtServices } = require('../services/jwt.services');
+const { tokenServices } = require('../services/token.services');
+const { usersServices } = require('../services/users.services');
+const {
+  sendActivationLink,
+  sendResetLink,
+} = require('../services/email.services');
+
+const { ApiError } = require('../exeptions/auth.error');
+
+const { checkPassword } = require('../utils/password/checkPassword');
+const { asyncHandler } = require('../utils/asyncHandler');
+
+const authValidation = async (req, _, next) => {
+  const { email, password } = req.body;
+
+  try {
+    await userSchema.validate({ email, password });
+  } catch (e) {
+    throw ApiError.badRequest(e.message);
+  }
+
+  req.userData = { email, password };
+  next();
+};
+
+const signUp = async (req, res) => {
+  const user = !!(await usersServices.getUserByEmail(req.userData.email));
+
+  if (user) {
+    throw ApiError.badRequest('User already exists');
+  }
+
+  const { email, activationToken } = await authServices.signUp(req.userData);
+
+  sendActivationLink(email, activationToken);
+
+  res
+    .status(201)
+    .json({ message: 'You need to activate your account, check your email' });
+};
+
+const signIn = async (req, res) => {
+  const { email, password } = req.userData;
+
+  const user = await usersServices.getUserByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  if (user.activationToken) {
+    throw ApiError.badRequest('Need to activate an account, check your email');
+  }
+
+  if (!(await checkPassword(password, user.password))) {
+    throw ApiError.badRequest('Password is wrong');
+  }
+
+  const userWithToken = await jwtServices.getUserWithToken(res, user);
+
+  await tokenServices.saveToken(user.id, userWithToken.refreshToken);
+
+  res.status(200).json(userWithToken);
+};
+
+const logOut = async (req, res) => {
+  const { refreshToken } = req;
```

Comment:
The `refreshToken` should be accessed from `req.cookies` instead of `req`. This is necessary to ensure that the token is correctly retrieved and cleared from the cookies.